### PR TITLE
Merge error codes from ebpf_windows.h into ebpf_result_t

### LINKED ISSUE #151: Merge ebpf error codes in ebpf_result.h and ebpf_windows.h in one enum / file.
Also, remove error codes from ebpf_windows.h file. Since these error codes are returned by public APIs, they should not be Windows specific.

### PR DESCRIPTION
**Pull request by dthaler**
Any values that seemed to duplicate entries already in ebpf_result_t were replaced with the existing values.
Any values that had a different meaning from values already in ebpf_result_t were added with the original names.
Any whitespace changes are from the format-code pass.

Fixes #151

Signed-off-by: Dave Thaler <dthaler@ntdev.microsoft.com>

### CODE REVIEW

File: include/ebpf_result.h [hunk 1]
```diff
@@ -43,14 +44,33 @@ extern "C"
         // Program or map is not pinned.
         EBPF_NOT_PINNED,
```
[the remark this answers is not in the document]
Comment (dthaler) [reply include/ebpf_result.h:45]: I was trying to not rename values in this PR, just to minimize changes (there's already a lot).
So would prefer to track any renames via a separate issue.

File: include/ebpf_result.h [hunk 2]
```diff
@@ -43,14 +44,33 @@ extern "C"
...
+        // Operation is not supported.
+        EBPF_ERROR_NOT_SUPPORTED,
+
+        // The requested item was not found.
+        EBPF_ERROR_NOT_FOUND,
+
+        // Access was denied for the requested operation.
+        EBPF_ERROR_ACCESS_DENIED,
+
+        EBPF_ERROR_BLOCKED_BY_POLICY,
```
[the remark this answers is not in the document]
Comment (dthaler) [reply include/ebpf_result.h:64]: done

File: include/ebpf_result.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (poornagmsft) [reply include/ebpf_result.h:46]: Sure. I have filed an issue. https://github.com/microsoft/ebpf-for-windows/issues/212

**Review by poornagmsft — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]
LGTM

**Pull request merged**